clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -50,15 +52,18 @@ class LakeFSRangeGetter(val apiConf: APIConfigurations, val configMapper: Config
         .map(o => new String(o.id))
   }
 
-  def getRangeAddresses(rangeID: String, repo: String): Iterator[String] = {
+  def getRangeAddresses(rangeID: String, repo: String, storageNS: String): Iterator[String] = {
     val location = ApiClient
       .get(apiConf)
       .getRangeURL(repo, rangeID)
     SSTableReader
       .forRange(configMapper.configuration, location)
       .newIterator()
-      .filter(_.message.addressType.isRelative)
-      .map(a => a.message.address)
+      .filter(e => e.message.addressType.isRelative || e.message.address.startsWith(storageNS))
```

Comment:
@johnnyaug Please note that this doesn't solve the issue for address type BY_PREFIX_DEPRECATED which is a relative path.
BY_PREFIX_DEPRECATED can theoretically have 3 type of addresses:

1. Relative path in namespace
2. Full path in namespace
3. Full path outside of namespace
You need to take all under consideration

@nopcoder is facing the same issue in https://github.com/treeverse/lakeFS/pull/4754

Reply:
I'm on with not solving it. It's been deprecated for a while now